src/Sql/Sql/ManagedInstance/Cmdlet/AzureSqlInstanceRefreshExternalGovernanceCmdletBase.cs

```diff
@@ -0,0 +1,44 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.Sql.Common;
+using Microsoft.Azure.Commands.Sql.ManagedInstance.Model;
+using Microsoft.Azure.Commands.Sql.ManagedInstance.Services;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Text;
+
+namespace Microsoft.Azure.Commands.Sql.ManagedInstance.Cmdlet
+{
+    public abstract class AzureSqlInstanceRefreshExternalGovernanceCmdletBase : AzureSqlCmdletBase<RefreshExternalGovernanceMIModel, RefreshExternalGovernanceAdapter>
+    {
+        /// <summary>
```

Comment:
Please add parameter sets if you have the time. You can check cmdlets like [SetAzureSqlManagedInstanceDtc.cs](https://github.com/Azure/azure-powershell/blob/main/src/Sql/Sql/ManagedInstance/Cmdlet/SetAzureSqlManagedInstance.cs) for some parameter set examples, and also there is an example in the docs:
https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/examples/set-cmdlet-example.md
Basically, the 2 additional parameter sets would allow the users to pass resourceId to the cmdlet, or to pass an MI object (which would be fetched by Get/List cmdlet for example)